nixos/modules/services/monitoring/librenms-agent.nix

```diff
@@ -0,0 +1,62 @@
+{ pkgs, config, lib, ... }:
+
+let
+  cfg = config.services.librenms-agent;
+in {
+  options.services.librenms-agent = {
+    enable = lib.mkEnableOption "LibreNMS agent";
+
+    package = lib.mkPackageOption pkgs "librenms-agent" {};
+
+    port = lib.mkOption {
+      type = lib.types.port;
+      default = 6556;
+      description = lib.mdDoc "Port where the LibreNMS Agent will listen.";
+    };
+
+    openFirewall = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Open port for LibreNMS Agent.
+      '';
+    };
+
+    ipAddressAllow = lib.mkOption {
+      example = [ "192.168.1.0/24" ];
+      type = lib.types.listOf lib.types.str;
+      description = ''
+        Allows access to the LibreNMS Agent only for the given addresses.
+      '';
+    };
+
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.sockets.librenms-agent = {
+      description = "Check_MK LibreNMS Agent Socket";
+      wantedBy = [ "sockets.target" ];
+      socketConfig = {
+        ListenStream = toString cfg.port;
+        Accept = "yes";
+      };
+    };
+
+    systemd.services."librenms-agent@" = {
+      description = "Check_MK LibreNMS Agent Service";
+      after = [ "network.target" "librenms-agent.socket" ];
+      requires = [ "librenms-agent.socket" ];
+      serviceConfig = {
+        ExecStart = lib.getExe cfg.package;
+        StandardOutput = "socket";
+        IPAddressDeny = "any";
+        IPAddressAllow = cfg.ipAddressAllow;
+      };
```

Comment:
```suggestion
      serviceConfig = lib.mkMerge [
        {
          ExecStart = lib.getExe cfg.package;
          StandardOutput = "socket";
        }
        (lib.mkIf (cfg.ipAddressAllow != []) {
          IPAddressDeny = "any";
          IPAddressAllow = cfg.ipAddressAllow;
        })
      ];
```

Reply:
@pbsds , should we really do this?

Please note what I mentioned initially.

> In addition, added customizable listenStream and ipAddressAllow options for slightly better security. Since the agent itself has no authentication, I've made ipAddressAllow mandatory.

If you think it doesn't bring any benefit to security, then we can proceed with this change.

Reply:
This is an _optional_ escape hatch, which must be explicitly configured by setting `ipAddressAllow = [];` since it has no default.

Reply:
Perhaps a comment in the `ipAddressAllow` option warning that a default value is explicitly omitted, is warranted.